ci/docker/linux-apt-jni.dockerfile

```diff
@@ -61,6 +61,7 @@ RUN apt-get update -y -q && \
         protobuf-compiler \
         rapidjson-dev \
         tzdata \
+        libcurl4-openssl-dev\
```

Comment:
Otherwise getting the following error when try `ARCH=amd64 docker-compose run debian-java-jni`:
```
Building AWS C++ SDK from source
CMake Error at /opt/cmake-3.11.4-Linux-x86_64/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /opt/cmake-3.11.4-Linux-x86_64/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /opt/cmake-3.11.4-Linux-x86_64/share/cmake-3.11/Modules/FindCURL.cmake:48 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake_modules/ThirdpartyToolchain.cmake:1087 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:4371 (find_curl)
  cmake_modules/ThirdpartyToolchain.cmake:4526 (build_awssdk)
  CMakeLists.txt:552 (include)
```